File: eTravelAgencija.Services/Database/OfferImage.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using eTravelAgencija.Services.Database;
using eTravelAgencija.Services.Services;

public class OfferImage
{
    [Key]
    public int Id { get; set; }

    [ForeignKey(nameof(OfferDetails))]
    public int OfferId { get; set; }  // FK prema OfferDetails

    public OfferDetails OfferDetails { get; set; }

    [Required]
    public string ImageUrl { get; set; }

    public bool isMain { get; set; }
}

File: UI/etravel_desktop/lib/models/offer_image.dart
class OfferImage {
  final int id;
  final int offerId;
  final String imageUrl;
  final bool isMain;

  OfferImage({
    required this.id,
    required this.offerId,
    required this.imageUrl,
    required this.isMain,
  });

  factory OfferImage.fromJson(Map<String, dynamic> json) {
    return OfferImage(
      id: json['id'],
      offerId: json['offerId'],
      imageUrl: json['imageUrl'],
      isMain: json['isMain'] ?? false,
    );
  }
}
